### PR TITLE
Bluetooth: samples: Update unicast server sample advertising #41329 

### PR DESCRIPTION
**Pull request by kaiser-ren**
According to BAP v1.0, section 3.5.3, unicast server shall transmit connectable extended advertising PDU with specific
AD format. 
This commit is trying to make unicast server being compatable with BAP v1.0 and use connectable extended advertising PDU, the AD format is:

- Length
- Type: «Service Data - 16-bit UUID» 
- Value: 
      ASCS UUID
      Announcement type: Target Announcement
      Available Audio Context
      Metadata Length = 0x00, non-metadata

### CODE REVIEW

File: samples/bluetooth/unicast_audio_server/prj.conf [hunk 1]
```diff
@@ -5,3 +5,6 @@ CONFIG_BT_AUDIO=y
 CONFIG_BT_AUDIO_UNICAST_SERVER=y
 # TODO: Remove, but is currently needed as UNICAST_SERVER cannot build on its own
 CONFIG_BT_AUDIO_UNICAST_CLIENT=y
+
+CONFIG_BT_EXT_ADV=y
+CONFIG_BT_DEVICE_NAME="Unicast Audio Server"
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/prj.conf:10]: Missing empty line

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 2]
```diff
@@ -25,12 +25,20 @@ NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, CONFIG_BT_ISO_TX_MTU, NULL);
 static struct bt_conn *default_conn;
 static struct bt_audio_stream streams[MAX_PAC];
 
+
+static uint8_t unicast_server_addata[] = {
+	0x4E, 0x18, /* ASCS UUID */
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:30]:
```suggestion
	BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL), /* ASCS UUID */
```

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 3]
```diff
@@ -25,12 +25,20 @@ NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, CONFIG_BT_ISO_TX_MTU, NULL);
...
+	0x01, /* Target Announcement */
+	0xFF, 0x00, 0xFF, 0x00, /* Available Audio Context */
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:32]: We should add `#define` for these values if they don't exist.

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 4]
```diff
@@ -25,12 +25,20 @@ NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, CONFIG_BT_ISO_TX_MTU, NULL);
...
+	0x00, /* Metadata length */
+};
+
 /* TODO: Expand with BAP data */
 static const struct bt_data ad[] = {
-	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:30]: We should still include these flags in the advertising data

File: samples/bluetooth/unicast_audio_server/boards/nrf52840dk_nrf52840.conf [hunk 5]
```diff
@@ -0,0 +1,2 @@
+CONFIG_BT_CTLR_ADV_EXT=y
+CONFIG_BT_CTLR_ADV_PERIODIC=y
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/boards/nrf52840dk_nrf52840.conf:2]: Can you please also add similar file for the `nrf5340dk_nrf5340_cpuapp` board (`nrf5340dk_nrf5340_cpuapp.conf`)?

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 6]
```diff
@@ -25,12 +25,20 @@ NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, CONFIG_BT_ISO_TX_MTU, NULL);
...
-	BT_DATA_BYTES(BT_DATA_UUID16_ALL, BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL)),
+	BT_DATA(BT_DATA_SVC_DATA16, unicast_server_addata, ARRAY_SIZE(unicast_server_addata)),
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:38]: Do we need to add more, or can the TODO in line 36 be removed now?

**Review by Thalley — CHANGES_REQUESTED** [review]

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 7]
```diff
@@ -25,12 +25,20 @@ NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, CONFIG_BT_ISO_TX_MTU, NULL);
...
+	0x4E, 0x18, /* ASCS UUID */
+	0x01, /* Target Announcement */
+	0xFF, 0x00, 0xFF, 0x00, /* Available Audio Context */
+	0x00, /* Metadata length */
+};
+
 /* TODO: Expand with BAP data */
 static const struct bt_data ad[] = {
-	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
-	BT_DATA_BYTES(BT_DATA_UUID16_ALL, BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL)),
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:31]: We should still advertise the `BT_DATA_UUID16_ALL` type with `BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL)`, even if we advertise the service data as well.

**Review by Thalley — CHANGES_REQUESTED** [review]

File: samples/bluetooth/unicast_audio_server/prj.conf (hunk 1 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/prj.conf:10]: Agree.

Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/prj.conf:10]: > Missing empty line

Agree.

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 3 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:32]: Yes, I agree that add #define for these Generic Audio Context Type, furthermore, Target/General Announcement. Which files do you think are suitable for bitmask definitions?

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 4 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:30]: > We should still include these flags in the advertising data

OK, will do.

File: samples/bluetooth/unicast_audio_server/boards/nrf52840dk_nrf52840.conf (hunk 5 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/boards/nrf52840dk_nrf52840.conf:2]: > Can you please also add similar file for the `nrf5340dk_nrf5340_cpuapp` board (`nrf5340dk_nrf5340_cpuapp.conf`)?

Will do. However, I don't have a 5340dk, thus can't verify it.

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 6 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:38]: > Do we need to add more, or can the TODO in line 36 be removed now?

I believe TODO can be removed even if some optimization can be taken, agree?

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 7 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:31]: > BT_DATA_UUID16_ALL

I don't think so. BT_DATA_UUID_ALL is 0x03, «Complete List of 16-bit Service Class UUIDs». However, BAP asks «Service Data - 16-bit UUID» which is 0x16. They are different.

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 3 above)
Comment (Thalley) [reply samples/bluetooth/unicast_audio_server/src/main.c:32]: It is a good question. We have a lot of missing definitions for LE Audio, and some of them are defined the profile specs or service specs, and some of them are defined by the Generic Audio Assigned Numbers. 

For now I would just put them in the audio.h file, and then possibly we'll move the BAP-only stuff from that file into other header files to clean it up (it is already very large).

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 6 above)
Comment (Thalley) [reply samples/bluetooth/unicast_audio_server/src/main.c:38]: Correct :)

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 7 above)
Comment (Thalley) [reply samples/bluetooth/unicast_audio_server/src/main.c:31]: Yes, but from 8.1.1.1 in BAP V 1.0, we also have 
image


So we should both advertise the flags, the service UUID AD data and the service data itself. It could of course be argued that if we advertise the service data from ASCS, that we could omit ASCS from the list of Service UUIDs.

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 3 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:32]: > It is a good question. We have a lot of missing definitions for LE Audio, and some of them are defined the profile specs or service specs, and some of them are defined by the Generic Audio Assigned Numbers.
> 
> For now I would just put them in the `audio.h` file, and then possibly we'll move the BAP-only stuff from that file into other header files to clean it up (it is already very large).

OK, I will put the definitions in audio.h temporarily.

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 7 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:31]: > Yes, but from 8.1.1.1 in BAP V 1.0, we also have image
> 
> So we should both advertise the flags, the service UUID AD data and the service data itself. It could of course be argued that if we advertise the service data from ASCS, that we could omit ASCS from the list of Service UUIDs.
Screen Shot 2021-12-28 at 10 32 30 PM
This is the AUX_ADV_IND which I'm trying to implement, I believe its Advertising Data Field is compatible with BAP definition, isn't it？ :)

Comment (Thalley) [reply samples/bluetooth/unicast_audio_server/src/main.c:31]: @kaiser-ren The flags looks correct, but it doesn't have the Service UUID AD data type as I pointed out. See https://www.bluetooth.com/specifications/specs/core-specification-supplement-10/ section 1.1

Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:31]: > @kaiser-ren The flags looks correct, but it doesn't have the Service UUID AD data type as I pointed out. See https://www.bluetooth.com/specifications/specs/core-specification-supplement-10/ section 1.1

I see, it looks like below.
Frame 22: 75 bytes on wire (600 bits), 75 bytes captured (600 bits) on interface /dev/cu.usbmodem14201-3.6, id 0
nRF Sniffer for Bluetooth LE
Bluetooth Low Energy Link Layer
    Access Address: 0x8e89bed6
    Packet Header: 0x3147 (PDU Type: AUX_ADV_IND, TxAdd: Random)
        .... 0111 = PDU Type: 0x7 AUX_ADV_IND
        ...0 .... = Reserved: 0
        ..0. .... = Reserved: 0
        .1.. .... = Tx Address: Random
        0... .... = Reserved: 0
        Length: 49
    Extended Advertising Header
        ..00 1001 = Extended Header Length: 9
        01.. .... = Advertising Mode: Connectable Non-scannable (0x1)
        Extended Header Flags: 0x09
        Advertising Address: d8:69:b1:a6:83:6e (d8:69:b1:a6:83:6e)
        Advertiser Data Info: 0x0e15
    Advertising Data
        Flags
            Length: 2
            Type: Flags (0x01)
            000. .... = Reserved: 0x0
            ...0 .... = Simultaneous LE and BR/EDR to Same Device Capable (Host): false (0x0)
            .... 0... = Simultaneous LE and BR/EDR to Same Device Capable (Controller): false (0x0)
            .... .1.. = BR/EDR Not Supported: true (0x1)
            .... ..1. = LE General Discoverable Mode: true (0x1)
            .... ...0 = LE Limited Discoverable Mode: false (0x0)
        16-bit Service Class UUIDs
            Length: 3
            Type: 16-bit Service Class UUIDs (0x03)
            UUID 16: Unknown (0x184e)
        Service Data - 16 bit UUID
            Length: 9
            Type: Service Data - 16 bit UUID (0x16)
            UUID 16: Unknown (0x184e)
            Service Data: 011f000f0000
        Device Name: Unicast Audio Server
            Length: 21
            Type: Device Name (0x09)
            Device Name: Unicast Audio Server
    CRC: 0x141247

File: include/bluetooth/audio.h [hunk 8]
```diff
@@ -36,6 +36,21 @@ struct bt_audio_cap;
...
+#define BT_AUDIO_CONTEXT_TYPE_MEDIA              0x0004
+#define BT_AUDIO_CONTEXT_TYPE_GAME               0x0008
+#define BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL      0x0010
+#define BT_AUDIO_CONTEXT_TYPE_VOICE_ASSISTANTS   0x0020
+#define BT_AUDIO_CONTEXT_TYPE_LIVE               0x0040
+#define BT_AUDIO_CONTEXT_TYPE_SOUND_EFFECTS      0x0080
+#define BT_AUDIO_CONTEXT_TYPE_NOTIFICATIONS      0x0100
+#define BT_AUDIO_CONTEXT_TYPE_RINGTONE           0x0200
+#define BT_AUDIO_CONTEXT_TYPE_ALERTS             0x0400
+#define BT_AUDIO_CONTEXT_TYPE_EMERGENCY_ALARM    0x0800
```
Comment (Thalley) [inline include/bluetooth/audio.h:52]: You can use the `BIT(x)` macro to make this a bit easier to read, e.g.

```
#define BT_AUDIO_CONTEXT_TYPE_PROHIBITED         0
#define BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED        BIT(1)
#define BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL     BIT(2)
```
etc.

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 9]
```diff
@@ -18,19 +18,39 @@
 
 #define MAX_PAC 1
 
+#define AVAILABLE_SINK_CONTEXT        (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
+BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | BT_AUDIO_CONTEXT_TYPE_MEDIA | \
+BT_AUDIO_CONTEXT_TYPE_GAME | BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL)
+#define AVAILABLE_SOURCE_CONTEXT      (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
+BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | BT_AUDIO_CONTEXT_TYPE_MEDIA | \
+BT_AUDIO_CONTEXT_TYPE_GAME)
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:26]:
```suggestion
#define AVAILABLE_SINK_CONTEXT  (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
				 BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | 
				 BT_AUDIO_CONTEXT_TYPE_MEDIA | \
				 BT_AUDIO_CONTEXT_TYPE_GAME | \
				 BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL)
				 
#define AVAILABLE_SOURCE_CONTEXT (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
				  BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | 
				  BT_AUDIO_CONTEXT_TYPE_MEDIA | \
				  BT_AUDIO_CONTEXT_TYPE_GAME)
```

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 10]
```diff
@@ -18,19 +18,39 @@
...
 static struct bt_audio_lc3_preset preset_16_2_1 = BT_AUDIO_LC3_UNICAST_PRESET_16_2_1;
 
 NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, CONFIG_BT_ISO_TX_MTU, NULL);
 static struct bt_conn *default_conn;
 static struct bt_audio_stream streams[MAX_PAC];
 
+
+static uint8_t unicast_server_addata[] = {
+	BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL), /* ASCS UUID */
+	0x01, /* Target Announcement */
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:38]: We should also have a `#define` for the target announcement

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 11]
```diff
@@ -18,19 +18,39 @@
...
+	(((AVAILABLE_SOURCE_CONTEXT) >>  0) & 0xFF),
+	(((AVAILABLE_SOURCE_CONTEXT) >>  8) & 0xFF),
+	0x00, /* Metadata length */
+};
+
 /* TODO: Expand with BAP data */
 static const struct bt_data ad[] = {
 	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
 	BT_DATA_BYTES(BT_DATA_UUID16_ALL, BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL)),
+	BT_DATA(BT_DATA_SVC_DATA16, unicast_server_addata, ARRAY_SIZE(unicast_server_addata)),
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:50]: I think you can also do something like 
```
BT_DATA_BYTES(BT_DATA_SVC_DATA16, unicast_server_addata)
```

**Review by Thalley — CHANGES_REQUESTED** [review]
A few minor comments, otherwise LGTM

File: include/bluetooth/audio.h (hunk 8 above)
Comment (kaiser-ren) [reply include/bluetooth/audio.h:52]: > You can use the `BIT(x)` macro to make this a bit easier to read, e.g.
> 
> ```
> #define BT_AUDIO_CONTEXT_TYPE_PROHIBITED         0
> #define BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED        BIT(1)
> #define BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL     BIT(2)
> ```
> 
> etc.

Great point, will do it.

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 10 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:38]: > We should also have a `#define` for the target announcement

Great, I will add the marco in audio.h

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 11 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:54]: > BT_DATA_SVC_DATA16

Just tried, I think BT_DATA_BYTES is good at for byte-wise operation, BT_DATA is for array-wise. Let's still keep BT_DATA.

Comment (Thalley) [reply samples/bluetooth/unicast_audio_server/src/main.c:54]: `BT_DATA_BYTES` is basically just adding the size of the content from `BT_DATA`. 

Normally you'd also use `sizeof` instead of `ARRAY_SIZE` for `BT_DATA`. It works right now since `unicast_server_addata` is a `uint8_t`, so `ARRAY_SIZE` returns the same value as `sizeof`. 

Its OK to keep, but `BT_DATA_BYTES` would just make this line slightly more simple.

File: samples/bluetooth/unicast_audio_server/src/main.c [hunk 12]
```diff
@@ -18,19 +18,43 @@
...
 };
 
+
```
Comment (Thalley) [inline samples/bluetooth/unicast_audio_server/src/main.c:57]: Remove empty line please :)

**Review by Thalley — APPROVED** [review]
It's OK to merge, but still 2 minor outstanding comments.

File: samples/bluetooth/unicast_audio_server/src/main.c (hunk 11 above)
Comment (kaiser-ren) [reply samples/bluetooth/unicast_audio_server/src/main.c:54]: > `BT_DATA_BYTES` is basically just adding the size of the content from `BT_DATA`.
> 
> Normally you'd also use `sizeof` instead of `ARRAY_SIZE` for `BT_DATA`. It works right now since `unicast_server_addata` is a `uint8_t`, so `ARRAY_SIZE` returns the same value as `sizeof`.
> 
> Its OK to keep, but `BT_DATA_BYTES` would just make this line slightly more simple.

I see, I tried some ways to test and the issue might be this:
below is BT_DATA_BYTES definition, you can see that it's friendly if _bytes are constant value, for example (0x12, 0x34, 0x56) and (uint8_t []) make it as an array type. I saw some declarations in different samples. 
However, if I put unicast_server_addata (an array name) instead of _bytes, it will cause a problem because ((uint8_t []) { unicast_server_addata } is incorrect. 

**#define BT_DATA_BYTES(_type, _bytes...) \
	BT_DATA(_type, ((uint8_t []) { _bytes }), \
		sizeof((uint8_t []) { _bytes }))**

Comment (Thalley) [reply samples/bluetooth/unicast_audio_server/src/main.c:54]: I see, so `BT_DATA_BYTES` doesn't work on variables, but only array literals? 

Fair enough, then `BT_DATA` works :)

File: include/bluetooth/audio.h [hunk 13]
```diff
@@ -36,6 +36,25 @@ struct bt_audio_cap;
...
+#define BT_AUDIO_CONTEXT_TYPE_LIVE               BIT(7)
+#define BT_AUDIO_CONTEXT_TYPE_SOUND_EFFECTS      BIT(8)
+#define BT_AUDIO_CONTEXT_TYPE_NOTIFICATIONS      BIT(9)
+#define BT_AUDIO_CONTEXT_TYPE_RINGTONE           BIT(10)
+#define BT_AUDIO_CONTEXT_TYPE_ALERTS             BIT(11)
+#define BT_AUDIO_CONTEXT_TYPE_EMERGENCY_ALARM    BIT(12)
+
+/* Unicast Announcement Type, Generic Audio */
+#define BT_AUDIO_UNICAST_ANNOUNCEMENT_GENERAL    0x00
+#define BT_AUDIO_UNICAST_ANNOUNCEMENT_TARGETED   0x01
```
Comment (jhedberg) [inline include/bluetooth/audio.h:56]: I don't think it's quite right to have a commit with "samples" in the title updating a public header file like this. This change is definitely related, and should therefore be part of the same PR, but it's really an independent dependency to the sample code changes, and should therefore be moved as a separate commit prior to the samples commit.

Comment (kaiser-ren) [reply include/bluetooth/audio.h:56]: > I don't think it's quite right to have a commit with "samples" in the title updating a public header file like this. This change is definitely related, and should therefore be part of the same PR, but it's really an independent dependency to the sample code changes, and should therefore be moved as a separate commit prior to the samples commit.

Agree, will have a separate commit for this.

**Comment by jhedberg** [comment]
@kaiser-ren looks pretty good now with how you've split up the commits, however their order is not right (which is not good e.g. when using `git bisect` - if you land in between the commits the sample would not build). Reversing them is quite trivial using e.g. `git bisect -i`. Could you also clean up the commit message of the samples commit, since it seems to contain redundant information:
```
Signed-off-by: Kai Ren <renkaikaiser@163.com>

Update samples/bluetooth/unicast_audio_server/src/main.c

Co-authored-by: Emil Gydesen <Thalley@users.noreply.github.com>

Update samples/bluetooth/unicast_audio_server/src/main.c

Co-authored-by: Emil Gydesen <Thalley@users.noreply.github.com>
```

It's ok to include a Co-authored-by line, but the Signed-off-by line of the commit author needs to be the last one (not sure if this is something that's enforced in Zephyr, but at least for the Linux kernel it is).

**Comment by kaiser-ren** [comment]
> @kaiser-ren looks pretty good now with how you've split up the commits, however their order is not right (which is not good e.g. when using `git bisect` - if you land in between the commits the sample would not build). Reversing them is quite trivial using e.g. `git rebase -i`. Could you also clean up the commit message of the samples commit, since it seems to contain redundant information:
> 
> ```
> Signed-off-by: Kai Ren <renkaikaiser@163.com>
> 
> Update samples/bluetooth/unicast_audio_server/src/main.c
> 
> Co-authored-by: Emil Gydesen <Thalley@users.noreply.github.com>
> 
> Update samples/bluetooth/unicast_audio_server/src/main.c
> 
> Co-authored-by: Emil Gydesen <Thalley@users.noreply.github.com>
> ```
> 
> It's ok to include a Co-authored-by line, but the Signed-off-by line of the commit author needs to be the last one (not sure if this is something that's enforced in Zephyr, but at least for the Linux kernel it is).

Good point and just fixed. 
I'm too new for git, if this fix pollutes anything or makes anything weird, apologize in advance.

**Review by jhedberg — APPROVED** [review]

**Comment by Thalley** [comment]
I'm merging this now. The documentation build issue is unrelated to this PR, and is not likely something we can easily fix, and the PDF documentation passes. 

This branch will also be merged into `main` very soon, and will get rid of whatever issue is causing it.
